### PR TITLE
NVSS-287 pagination

### PR DESCRIPTION
**Pull request by mitre-jim**

### CODE REVIEW

**Comment by smacadam** [comment]
When I tested this locally, the returned next links were 
    "link": [
        {
            "relation": "next",
            "url": "file:///MA/Bundles"
        }
    ]
and 
    "link": [
        {
            "relation": "first",
            "url": "file:///MA/Bundles%3F_since=06%252F16%252F2022%252015%253A28%253A01&page=1"
        },
        {
            "relation": "last",
            "url": "file:///MA/Bundles%3F_since=06%252F16%252F2022%252015%253A28%253A01&page=0"
        }
    ]

**Comment by smacadam** [comment]
url format has been fixed and README has been updated

**Comment by smacadam** [comment]
Made page size configurable

File: messaging.tests/Integration/BundlesControllerTests.cs [hunk 1]
```diff
@@ -265,6 +265,134 @@ public async System.Threading.Tasks.Task ReturnErrorOnInvalidBatch()
...
+          Bundle bundle2 = parser.Parse<Bundle>(bundleOfBundles2);
+          Assert.Equal(10, bundle2.Entry.Count);
+
+          // 4th response is 0 records
+          HttpResponseMessage getBundles3 = await JsonResponseHelpers.GetAsync(_client, "/MA/Bundles?_count=20");
+          Assert.Equal(HttpStatusCode.OK, getBundles3.StatusCode);
+
+          string bundleOfBundles3 = await getBundles3.Content.ReadAsStringAsync();
+          Bundle bundle3 = parser.Parse<Bundle>(bundleOfBundles3);
+          Assert.Equal(0, bundle3.Entry.Count);
```
Comment (pkmitre) [inline messaging.tests/Integration/BundlesControllerTests.cs:347]: Should be able to address this warning by replacing this line with

Assert.Empty(bundle3.Entry);

File: messaging/Controllers/BundlesController.cs [hunk 2]
```diff
@@ -29,19 +29,39 @@ public BundlesController(ILogger<BundlesController> logger, ApplicationDbContext
...
+                return BadRequest("_count must not be negative");
+            }
+            if (page < 1)
+            {
+                return BadRequest("page must not be negative");
+            }
+            // Retrieving unread messages changes the result set (as they get marked read), so we don't REALLY support paging
+            if (_since == default(DateTime) && page > 1)
+            {
+                return BadRequest("Paging not supported");
```
Comment (pkmitre) [inline messaging/Controllers/BundlesController.cs:62]: I think the message here could be clearer, maybe something like "Pagination does not support specifying a page other than the immediate next available page"

**Review by pkmitre — APPROVED** [review]

**Pull request merged**